plugins/status/plugin.go

```diff
@@ -90,16 +94,17 @@ func ParseConfig(config []byte, services []string) (*Config, error) {
 }
 
 // New returns a new Plugin with the given config.
-func New(parsedConfig *Config, manager *plugins.Manager) *Plugin {
+func New(parsedConfig *Config, manager *plugins.Manager, globalMetrics metrics.GlobalMetrics) *Plugin {
```

Comment:
nit: If we wanted to keep these public golang APIs from changing we probably need to leave the constructor alone and add a helper function to set the globalMetrics on the plugin. That being said I don't have a good feel for who all might be using them.. so maybe its non-issue.

Reply:
changed it to `New(..., ...func(*Plugin)) pattern instead that doesn't break the signature/existing consumers